docs/lockfile_generation.md

```diff
@@ -1,9 +1,17 @@
 # Lockfile Generation
 
-The Phylum CLI can generate a lockfile when it is given a manifest file.
+Lockfiles can be directly parsed and analyzed by Phylum's CLI. However since
+manifest files only specify loose version requirements, it is necessary for the
+CLI to internaly generate the corresponding lockfile.
+
+No lockfile generation will take place if the `--no-generation` flag is passed
+to `parse` or `analyze`.
```

Comment:
Previously this was under the "lockfile detection" heading, but it doesn't really have anything to do with lockfile detection.